superdesk/media/video_editor.py

```diff
@@ -0,0 +1,180 @@
+
+    def get_paginate(self, offset):
+        """Get list of projects.
+
+        :param offset:
+        """
+        raise NotImplementedError
+
+    def post_project(self, filestream):
+        """Create new project.
+
+        :param filestream:
+        """
+        raise NotImplementedError
+
+    def post(self, project_id, updates):
+        """Edit video. This method creates a new project.
+
+        :param project_id:
+        :param updates: changes apply to the video
+        """
+        raise NotImplementedError
+
+    def put(self, project_id, updates):
+        """Edit video. This method does not create a new project.
+
+        :param project_id:
+        :param updates: changes apply to the video
+        """
+        raise NotImplementedError
+
+    def delete(self, project_id):
+        """Delete video in video server.
+
+        :param project_id:
+        """
+        raise NotImplementedError
+
+    def get_timeline_thumbnails(self, project_id, amount):
+        """Get video thumbnails.
+
+        :param project_id:
+        :param amount: number of thumbnails to generate/get
+        """
+        raise NotImplementedError
+
+    def post_preview_thumbnail(self, project_id, post_type, time=None, base64_data=None):
+        """Upload video preview thumbnail to video server.
+
+        :param project_id:
+        :param post_type: Post preview thumbnail type, either capture or upload
+        :param time: time to capture
+        :param base64_data: base64 image data if type is upload
+        """
+        raise NotImplementedError
+
+    def get_preview_thumbnail(self, project_id, time=None):
+        """Capture image from video-server.
+
+        :param project_id:
+        :param time: period time to capture the image in video
+        :return:
+        """
+        raise NotImplementedError
+
+    def check_video_server(self):
+        """Check the video-server is exist."""
+        raise NotImplementedError
+
+
+class VideoEditorService(VideoEditorFactory):
+    """Video Service for integating with video server"""
+
+    label = 'Video Editor'
+
+    def __init__(self, ):
+        self.session = requests.Session()
+        self.session.headers.update({'User-Agent': 'superdesk'})
+
+    def get_base_url(self):
+        return app.config.get('VIDEO_SERVER_HOST') + '/projects/'
+
+    def check_video_server(self):
+        parse = urlparse(app.config.get('VIDEO_SERVER_HOST'))
+        host = parse.hostname
+        port = parse.port if parse.port else 80
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        result = sock.connect_ex((host, port))
+        if result == 0:
+            return True
+        else:
+            return False
+
+    def _url(self, _id, resource=''):
+        url = urljoin(self.get_base_url(), _id)
+        if resource:
+            return urljoin(url + '/', resource)
+        return url
+
+    def _get_response(self, resp, *status):
+        json = json_util.loads(resp.text)
```

Comment:
I guess this might fail for some network errors, maybe it should only do it for 400 errors or good responses? or use try/catch